Userland/Games/BrickGame/BrickGame.cpp

```diff
@@ -0,0 +1,554 @@
+        switch (num_removed_rows) {
+        case 1:
+            m_score += 40 * (m_level + 1);
+            break;
+        case 2:
+            m_score += 100 * (m_level + 1);
+            break;
+        case 3:
+            m_score += 300 * (m_level + 1);
+            break;
+        case 4:
+            m_score += 1200 * (m_level + 1);
+            break;
+        default:
+            break;
+        }
+    }
+};
+
+BrickGame::BrickGame(StringView app_name)
+    : m_app_name { app_name }
+    , m_state { GameState::Idle }
+    , m_brick_game(make<Bricks>())
+{
+    set_font(Gfx::FontDatabase::default_fixed_width_font().bold_variant());
+    m_high_score = Config::read_i32(m_app_name, m_app_name, "HighScore"sv, 0);
+    reset();
+}
+
+void BrickGame::reset()
+{
+    m_state = GameState::Active;
+    m_brick_game->reset();
+    stop_timer();
+    start_timer(15); // 66.6ms
+    if (!m_brick_game->new_block())
+        VERIFY_NOT_REACHED();
+    update();
+}
+
+void BrickGame::timer_event(Core::TimerEvent&)
+{
+    if (m_brick_game->is_game_over()) {
+        game_over();
+        return;
+    }
+    m_brick_game->update();
+    update(); // TODO: optimize repainting
+}
+
+void BrickGame::keydown_event(GUI::KeyEvent& event)
+{
+    switch (event.key()) {
+    case KeyCode::Key_A:
+    case KeyCode::Key_H:
+    case KeyCode::Key_Left:
+        m_brick_game->move_left();
+        break;
+    case KeyCode::Key_D:
+    case KeyCode::Key_L:
+    case KeyCode::Key_Right:
+        m_brick_game->move_right();
+        break;
+    case KeyCode::Key_W:
+    case KeyCode::Key_K:
+    case KeyCode::Key_Up:
+        m_brick_game->rotate_right();
+        break;
+    case KeyCode::Key_E:
+        m_brick_game->rotate_left();
+        break;
+    case KeyCode::Key_S:
+    case KeyCode::Key_Down:
+        m_brick_game->move_down();
+        break;
+    case KeyCode::Key_Space:
+        m_brick_game->move_down_fast();
+        break;
+    default:
+        break;
+    }
+}
+
+void BrickGame::paint_cell(GUI::Painter& painter, Gfx::IntRect rect, bool is_on)
+{
+    painter.draw_rect(rect, m_back_color);
+    rect.inflate(-1, -1, -1, -1);
+    painter.draw_rect(rect, is_on ? m_front_color : m_shadow_color);
+    painter.set_pixel(rect.top_left(), m_back_color);
+    painter.set_pixel(rect.bottom_left(), m_back_color);
+    painter.set_pixel(rect.top_right(), m_back_color);
+    painter.set_pixel(rect.bottom_right(), m_back_color);
+    rect.inflate(-2, -2);
+    painter.draw_rect(rect, is_on ? m_front_color : m_shadow_color);
+    rect.inflate(-2, -2);
+    painter.draw_rect(rect, m_back_color);
+    rect.inflate(-2, -2);
+    painter.draw_rect(rect, m_back_color);
+    rect.inflate(-2, -2);
+    painter.fill_rect(rect, is_on ? m_front_color : m_shadow_color);
+}
+
+void BrickGame::paint_text(GUI::Painter& painter, int row, String const& text)
+{
+    auto const text_width { font().width(text) };
+    auto const r { frame_inner_rect() };
```

Comment:
This could be given a more descriptive name, such as `frame_inner_rect_size` or similar.